runtime/lua/vim/diagnostic.lua

```diff
@@ -1129,7 +1129,10 @@ end
 --- Limit diagnostics to one or more namespaces.
 --- @field namespace? integer[]|integer
 ---
---- Limit diagnostics to the given line number.
+--- Limit diagnostics to those spanning the specified line number.
+--- Diagnostics are included if the specified line number falls between
+--- the `lnum` and `end_lnum` of the diagnostic. If `end_lnum` is not provided,
+--- it is assumed to be equal to `lnum`.
```

Comment:
"Provided" here implies that `end_lnum` is given in the options, rather than something that exists in the diagnostic. 

Personally I don't think we need to leak implementation details here, "spanning the specified line number" is sufficient to describe how this works.

```suggestion
```

Reply:
removed.